File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }
        
        List<Integer> counts = new ArrayList<>(map.values());
        
        counts.sort(Collections.reverseOrder());
        
        int answer = 0;
        
        for (int count : counts) {
            k -= count;
            answer++;
            
            if (k <= 0) {
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    unordered_map<int, int> counts;
    for (int size : tangerine) {
        counts[size]++;
    }
    
    vector<int> freq;
    for (auto const& [size, count] : counts) {
        freq.push_back(count);
    }
    
    sort(freq.begin(), freq.end(), greater<int>());
    
    for (int count : freq) {
        k -= count;
        answer++;
        
        if (k <= 0) {
            break;
        }
    }
    return answer;
}